docs/visual_studio_code.md

```diff
@@ -0,0 +1,230 @@
+
+In this example you can see that it shows that a `str` of `'23'` is not a valid `int` for the argument `age`.
+
+![VS Code strict type errors](./img/vs_code_06.png)
+
+It would expect `age=23` instead of `age='23'`.
+
+Nevertheless, the design, and one of the main features of *pydantic*, is that it is very **lenient with data types**.
+
+It will actually accept the `str` with value `'23'` and will convert it to an `int` with value `23`.
+
+These strict error checks are **very useful** most of the time and can help you **detect many bugs early**. But there are cases, like with `age='23'`, where they could be inconvenient by reporting a "false positive" error.
+
+---
+
+This example above with `age='23'` is intentionally simple, to show the error and the differences in types.
+
+But more common cases where these strict errors would be inconvenient would be when using more sophisticated data types, like `int` values for `datetime` fields, or `dict` values for *pydantic* sub-models.
+
+For example, this is valid for *pydantic*:
+
+```Python hl_lines="12 17"
+from pydantic import BaseModel
+
+
+class Knight(BaseModel):
+    title: str
+    age: int
+    color: str = 'blue'
+
+
+class Quest(BaseModel):
+    title: str
+    knight: Knight
+
+
+quest = Quest(
+    title='To seek the Holy Grail',
+    knight={'title': 'Sir Lancelot', 'age': 23}
+)
+```
+
+The type of the field `knight` is declared with the class `Knight` (a *pydantic* model) and the code is passing a literal `dict` instead. This is still valid for *pydantic*, and the `dict` would be automatically converted to a `Knight` instance.
+
+Nevertheless, it would be detected as a type error:
+
+![VS Code strict type errors with model](./img/vs_code_07.png)
+
+#### Disable type checks in a line
+
+In those cases, you can disable the errors for a specific line using a comment of:
+
+```
+# type: ignore
+```
+
+coming back to the example with `age='23'`, it would be:
+
+```Python hl_lines="10"
+from pydantic import BaseModel
+
+
+class Knight(BaseModel):
+    title: str
+    age: int
+    color: str = 'blue'
+
+
+lancelot = Knight(title='Sir Lancelot', age='23')  # type: ignore
+```
+
+that way Pylance/Pyright and mypy will ignore errors in that line.
+
+#### Override the type of a variable
+
+Alternatively, you can create a variable with the value you want to use, and set an explicit type of `Any`.
+
+```Python hl_lines="1 11-12"
+from typing import Any
+from pydantic import BaseModel
+
+
+class Knight(BaseModel):
+    title: str
+    age: int
+    color: str = 'blue'
+
+
+age_str: Any = '23'
+lancelot = Knight(title='Sir Lancelot', age=age_str)
+```
+
+that way Pylance/Pyright and mypy will interpret the variable `age_str` as if they didn't know its type, instead of knowing it has a type of `str` when an `int` was expected (and then showing the corresponding error).
+
+The advantage of this technique is that you will still see any additional errors for the other arguments.
```

Comment:
Well, the advantage is that it will not disable all type checking for that line, e.g. it will alert you if you misspell `age` or the name of the class or if you make a syntactical error.  `# type: ignore` disables all of these checks.  A third option would also be to use `cast`.

Reply:
Good point. And great idea! `cast` could work quite well. 🚀

I'll add the `cast` option and update the docs for that to explain the disadvantages of `# type: ignore` as well. And show the 3 alternatives in order of increasing steps/complexity.

Reply:
`cast` should work nicely but as I mentioned in the discussion, the problem is that is technically wrong.

`cast()` says "this think is actually an `int`, promise. If I'm wrong, it's my problem" here that's not the case, the thing really is a string (or whatever) it's just that that type is valid.

I'm afraid sadly there's no non-hack here, but I guess `cast()` could be useful.

Reply:
The non-hack is to be explicit in type conversions. If pydantic contains logic that performs type conversions implicitly, does it expose those type conversion routines so they can be invoked explicitly for users who are interested in static type safety? The use of `Any` or `cast` or `# type: ignore` are all poor workarounds if you care about static type checking.

Reply:
> `cast` should work nicely but as I mentioned in the discussion, the problem is that is technically wrong. [...]

Yep, agreed. I updated the docs including another example using `cast(Any, '23')` instead of `cast(int, '23')`, I think that's an acceptable balance, telling the editor "don't check this", which would also work for any other type, without having to do something that is technically wrong like `cast(int, '23')`.

> The non-hack is to be explicit in type conversions. [...]

Agreed, and for cases like the example with a `'23'` it's quite obvious and it would be a lot better to do the conversion manually.

But for things like [`datetime`s](https://pydantic-docs.helpmanual.io/usage/types/#datetime-types) that accept multiple values including `str`, `int`, `float`, or for passing literal `dict`s in places declared with a *pydantic* model class, I think that using a single `cast()` could be an acceptable tradeoff for now.

> [...] does it expose those type conversion routines so they can be invoked [...]

I understand that not currently. Maybe that could be a new feature request, and then these docs could be updated accordingly. But I think that with respect to this PR, these docs could be enough for now.